geonode/qgis_server/views.py

```diff
@@ -185,8 +185,12 @@ def download_map(request, mapid):
     zf.close()
 
     # Grab ZIP file from in-memory, make response with correct MIME-type
-    resp = HttpResponse(
-        s.getvalue(), content_type="application/x-zip-compressed")
+    try:
+        resp = HttpResponse(
+            s.getvalue(), content_type="application/x-zip-compressed")
+    # TODO: we expect a MemoryError on downloading huge files
+    except Exception as e:
```

Comment:
Why not put MemoryError here since the error message is about files too big?

Reply:
i tried to generate MemoryError on local machine, but failed.
the reason i use wider exception here is because this part hasn't verified yet.
even with this Exception, the local server just stopped without any warning.

that's why i put a comment.
perhaps we could try it on testing?

Reply:
thanks @ismailsunni. i've changed accordingly.